### PR TITLE
User can specific simulate sample's message sending period

### PR DESCRIPTION
**Pull request by yuwzho**
The simulate sample always send a message every 10 seconds in currect time. 

It is different with the BLE sample, which will make the "simlate" sample not simulate the BLE sample's behavior.

### CODE REVIEW

**Comment by yuwzho** [comment]
@andrew-buckley

File: modules/simulated_device/src/simulated_device.c [hunk 1]
```diff
@@ -190,7 +199,7 @@ static MODULE_HANDLE SimulatedDevice_Create(BROKER_HANDLE broker, const void* co
 
 static void * SimulatedDevice_ParseConfigurationFromJson(const char* configuration)
```
Comment (darobs) [inline modules/simulated_device/src/simulated_device.c]: Should there be corresponding changes to FreeConfigurationFromJson?

File: modules/simulated_device/src/simulated_device.c [hunk 2]
```diff
@@ -222,10 +232,31 @@ static void * SimulatedDevice_ParseConfigurationFromJson(const char* configurati
...
+                        if (mallocAndStrcpy_s(&(config.macAddress), macAddress) != 0)
+                        {
+                            result = NULL;
+                        }
+                        else
+                        {
+                            config.messagePeriod = period;
+                            result = malloc(sizeof(SIMULATEDDEVICE_CONFIG));
+                            if (result == NULL) {
+                                LogError("allocation of configuration failed");
```
Comment (darobs) [inline modules/simulated_device/src/simulated_device.c]: Free config.macAddress as well, please.

**Comment by darobs** [comment]
I'm sorry, @yuwzho, I noticed too late that this was a branch off of master, would you please make a pull request from develop?

**Comment by yuwzho** [comment]
@darobs Sure, I will fix your comments and send a new pullrequest to the develop branch

**Comment by yuwzho** [comment]
Open another one to develop branch and the new one already refers to this one. So close it.

**Pull request closed without merging**